src/cakes/search.rs

```diff
@@ -170,8 +166,8 @@ impl<T: Number, U: Number> Search<T, U> {
 mod tests {
     use std::sync::Arc;
 
-    use crate::dataset::Dataset;
-    use crate::dataset::RowMajor;
+    use crate::clam::Dataset;
+    use crate::clam::dataset::RowMajor;
```

Comment:
I'm curious why dataset is falling into the clam mod

Reply:
Dataset is part of the core framework though I could be persuaded to treat Dataset (and Metric) differently. Dataset and metric undergird Cluster, Graph, and Manifold